packages/libp2p/test/connection-manager/connection-pruner.spec.ts

```diff
@@ -0,0 +1,98 @@
+import { TypedEventEmitter } from '@libp2p/interface'
+import { defaultLogger } from '@libp2p/logger'
+import { PeerMap } from '@libp2p/peer-collections'
+import { createEd25519PeerId } from '@libp2p/peer-id-factory'
+import { expect } from 'aegir/chai'
+import { stubInterface } from 'sinon-ts'
+import { ConnectionPruner } from '../../src/connection-manager/connection-pruner.js'
+import type { Libp2pEvents, PeerStore, Stream, TypedEventTarget, Connection } from '@libp2p/interface'
+import type { ConnectionManager } from '@libp2p/interface-internal'
+import type { StubbedInstance } from 'sinon-ts'
+
+interface ConnectionPrunerComponents {
+  connectionManager: StubbedInstance<ConnectionManager>
+  peerStore: StubbedInstance<PeerStore>
+  events: TypedEventTarget<Libp2pEvents>
+}
+
+describe('connection-pruner', () => {
+  let pruner: ConnectionPruner
+  let components: ConnectionPrunerComponents
+
+  beforeEach(() => {
+    components = {
+      connectionManager: stubInterface<ConnectionManager>(),
+      peerStore: stubInterface<PeerStore>(),
+      events: new TypedEventEmitter()
+    }
+
+    pruner = new ConnectionPruner({
+      ...components,
+      logger: defaultLogger()
+    }, {})
+  })
+
+  it('should sort connections for pruning', async () => {
```

Comment:
I like the succinctness of this test but I do feel for readability it is at times better to be more explicit (at the expense of repetition). So a test would probably read as:

_older connections with streams should be closed before tagged connections_ for example. 

Otherwise debugging the sorting algorithm could become a bit cumbersome.